Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -388,7 +388,8 @@ public class PBXProjGenerator {
 
             if target.type.isApp && target.platform != .macOS {
                 let inputPaths = Set(carthageFrameworksToEmbed).map { "$(SRCROOT)/\(carthageBuildPath)/\(target.platform)/\($0)\($0.contains(".") ? "" : ".framework")" }
-                let carthageScript = PBXShellScriptBuildPhase(reference: generateUUID(PBXShellScriptBuildPhase.self, "Carthage" + target.name), files: [], name: "Carthage", inputPaths: inputPaths, outputPaths: [], shellPath: "/bin/sh", shellScript: "/usr/local/bin/carthage copy-frameworks\n")
+                let outputPaths = Set(carthageFrameworksToEmbed).map { "$(BUILT_PRODUCTS_DIR)/$(FRAMEWORKS_FOLDER_PATH)/\($0)\($0.contains(".") ? "" : ".framework")" }
```

Comment:
Good catch @mironal. This helps xcode now when it needs to do recompile certain components because any of the frameworks changed. Looks good to me!

Reply:
@mironal I believe the input and output paths have to line up in terms of their order. The code here SHOULD be fine, but the lists may be out of order. That's because Set doesn't guarantee an order, though I think it would be the same with the same data and same run.
To be safe can you remove the Set here and move it up to the carthageFrameworksToEmbed?

```
let carthageFrameworksToEmbed = Array(Set(carthageDependencies
            .filter { ($0.embed ?? true) }
            .map { $0.reference }))
            .sorted()
```
The Set is used by the way to remove duplicates, as an application target as well it's dependent targets may reference the same carthage dependency

Reply:
Thanks for your code review. I will fix it. 😄